#!/bin/sh -e
## 149_pr11456.dpatch
##
## DP: Description: PR gas/11456: Use memcpy to copy overlap memory.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

2010-03-30  H.J. Lu  <hongjiu.lu@intel.com>
	
	PR gas/11456
	* input-scrub.c (input_scrub_next_buffer): Use memmove instead
	of memcpy to copy overlap memory.

@DPATCH@
===================================================================
RCS file: /cvs/src/src/gas/input-scrub.c,v
retrieving revision 1.22
retrieving revision 1.23
diff -u -r1.22 -r1.23
--- src/gas/input-scrub.c	2009/09/11 15:27:33	1.22
+++ src/gas/input-scrub.c	2010/03/30 23:20:25	1.23
@@ -343,8 +343,8 @@
 
   if (partial_size)
     {
-      memcpy (buffer_start + BEFORE_SIZE, partial_where,
-	      (unsigned int) partial_size);
+      memmove (buffer_start + BEFORE_SIZE, partial_where,
+	       (unsigned int) partial_size);
       memcpy (buffer_start + BEFORE_SIZE, save_source, AFTER_SIZE);
     }
   limit = input_file_give_next_buffer (buffer_start
